### PR TITLE
Allow user to select vhost priority

### PR DESCRIPTION
**Pull request by lazyfrosch**
Please let me know if you would accept this. I'd add a spec test if you like.

### CODE REVIEW

**Comment by mmoll** [comment]
could you describe your use case?

**Comment by lazyfrosch** [comment]
The idea is to have a server, that is not purely dedicated to Puppet or Foreman, so I still want a properly configured default vhost for the $::fqdn, while Foreman resides on a special servername like `foreman.example.com`.

The Puppet vhost doesn't matter, since its put on a extra port.

I just missed the flexibility here, and had to hand fiddle with the default vhosts.

**Comment by mmoll** [comment]
Ah, mainly http, I'd suggerst to name the parameter `vhost_priority`, no objections otherwise.

**Comment by lazyfrosch** [comment]
Yeah that makes sense, I updated my commit.

**Comment by lazyfrosch** [comment]
Uhm the errornous spec tests dont seem related to my change.

**Comment by ekohl** [comment]
It looks like you're right: https://travis-ci.org/theforeman/puppet-foreman/jobs/119093494 fails with the same error.

File: manifests/config/passenger.pp [hunk 1]
```diff
@@ -55,6 +57,7 @@
   $app_root               = $::foreman::app_root,
   $listen_on_interface    = $::foreman::passenger_interface,
   $ruby                   = $::foreman::passenger_ruby,
+  $priority               = $::foreman::vhost_priority,
```
Comment (ekohl) [inline manifests/config/passenger.pp]: Could you add a ```validate_string``` for this?

**Comment by mmoll** [comment]
please rebase onto current master to get fixed tests.

**Comment by lazyfrosch** [comment]
Rebased and with validate_string

**Comment by ekohl** [comment]
:+1: on green tests.

**Comment by mmoll** [comment]
merged, Danke @lazyfrosch!

**Pull request closed without merging**